src/Workspaces/CSharp/Portable/Simplification/Reducers/AbstractCSharpReducer.cs

```diff
@@ -14,5 +14,11 @@ internal abstract partial class AbstractCSharpReducer : AbstractReducer
         protected AbstractCSharpReducer(ObjectPool<IReductionRewriter> pool) : base(pool)
         {
         }
+
+        public sealed override bool IsApplicable(SimplifierOptions options)
+            => IsApplicable((CSharpSimplifierOptions)options);
```

Comment:
consider making AbstractReducer generic on TSimplifierOptions.

Reply:
I did, but it gets quite complicated since we have a lot of code that operates on AbstractReducer that's not specialized. So we would need to switch that code to an interface and effectively just moved the cast somewhere else.